types/helpers.d.ts

```diff
@@ -20,14 +20,13 @@ declare module 'mongoose' {
    * of an ObjectId.
    */
   function isObjectIdOrHexString(v: mongodb.ObjectId): true;
-  function isObjectIdOrHexString(v: string): boolean;
+  function isObjectIdOrHexString(v: mongodb.ObjectId | string): boolean;
```

Comment:
Welcome to mongoose @10kc-emilyk 

Shouldn't `mongodb.ObjectId` always be true, since it is an object id?

Also, I wonder what effect will that have on the typing, if we have one signature with `ObjectId`, and another with `ObjectId | string`. Which one will TS pick when I use `isObjectIdOrHexString(someObjectId);`?

Reply:
TS will pick the first overload signature, so when the type is just `ObjectId`, the return type will resolve as `true`. The secondary one will only be hit if you're passing a value that could be an objectId _or_ a string. Currently, if you pass a value that is `ObjectId | string`, the return type will be `false`.

Reply:
Thanks for the explanation.

Does that mean the proposed change below for `isValidObjectId` will make it so that all the following cases will return `true`?
* mongodb.ObjectId | Types.ObjectId
* mongodb.ObjectId
* Types.ObjectId

Or will it only return true if the type is an exact match (e.g. `mongodb.ObjectId | Types.ObjectId`)?

Can we add some type tests to /tests/types to verify that things work well and avoid regressions in the future?

Reply:
We have already typings tests.

Reply:
Today I learned: you can see references for types from PRs directly:
<img width="896" alt="Screenshot 2022-06-10 at 5 00 04 PM" src="https://user-images.githubusercontent.com/19984935/173094247-272579c4-2998-483d-80d9-04dec82673c1.png">
Thanks @Uzlopak